AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -25,6 +26,41 @@ static void runLoopSourceCallback(void *info) {
 #endif
 }
 
+#pragma mark - ASDeallocThread
+
+@interface ASDeallocThread : NSThread
+@property (nonatomic, strong) ASRunLoopQueue *deallocQueue;
+@end
+
+@implementation ASDeallocThread
+
+- (void)main
+{
+  self.name = @"ASDeallocThread";
+  [self deallocQueue];
+  while (YES) {
+    // This method will still return once all sources have finished.
+    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate distantFuture]];
```

Comment:
Did you look into `ASRunLoopQueue` how we handle it? Especially look at the `_runLoopObserver` / `CFRunLoopObserverRef`. I know it's not a subclass of `NSThread` but I think you can model it similar to it.